rpcs3/Emu/Cell/SPUASMJITRecompiler.cpp

```diff
@@ -1221,15 +1223,103 @@ void spu_recompiler::RCHCNT(spu_opcode_t op)
 	}
 	case SPU_RdEventStat:
 	{
-		// TODO
-		break;
+		LOG_ERROR(SPU, "SPU_RdEventStat");
+		Label after_rcheck = c->newLabel();
+		Label fallback = c->newLabel();
+		Label rfail = c->newLabel();
+		Label next = c->newLabel();
+		c->mov(*addr, SPU_OFF_32(raddr));
+		c->test(*addr, *addr);
+		c->jz(after_rcheck);
+		c->mov(*qw0, imm_ptr(vm::g_reservations));
+		c->shr(*addr, 4);
+		c->mov(*qw0, x86::qword_ptr(*qw0, *addr));
+		c->cmp(*qw0, SPU_OFF_64(rtime));
+		c->jne(rfail);
+		c->mov(*qw0, imm_ptr(vm::g_base_addr));
+		if (utils::has_avx())
+		{
+			c->vmovaps(x86::ymm0, x86::yword_ptr(*qw0, *addr, 0));
+			c->vxorps(x86::ymm1, x86::ymm0, x86::yword_ptr(*cpu, offset32(&SPUThread::rdata) + 0));
+			c->vmovaps(x86::ymm0, x86::yword_ptr(*qw0, *addr, 32));
+			c->vxorps(x86::ymm2, x86::ymm0, x86::yword_ptr(*cpu, offset32(&SPUThread::rdata) + 32));
+			c->vmovaps(x86::ymm0, x86::yword_ptr(*qw0, *addr, 64));
+			c->vxorps(x86::ymm3, x86::ymm0, x86::yword_ptr(*cpu, offset32(&SPUThread::rdata) + 64));
+			c->vmovaps(x86::ymm0, x86::yword_ptr(*qw0, *addr, 96));
+			c->vxorps(x86::ymm4, x86::ymm0, x86::yword_ptr(*cpu, offset32(&SPUThread::rdata) + 96));
+			c->vorps(x86::ymm0, x86::ymm1, x86::ymm2);
```

Comment:
ymm0 still contains a part of the lock line data, it needs to be cleared before this line for the check to work (not sure, i may be wrong)